"Scroll an element into view" in HTML spec

nanto opened the pull request:
This is tests for whatwg/html#3131.

(I'm advised to write tests in https://github.com/whatwg/html/issues/3641#issuecomment-383603754)

<!-- Reviewable:start -->

<!-- Reviewable:end -->

nanto committed:
scrollPathIntoView method of CanvasRenderingContext2D

I use assert_approx_equals here since properties of DOMRect are of type double.

nanto committed:
Navigating to a fragment in HTML spec

zcorpan commented:
html/browsers/browsing-the-web/scroll-to-fragid/scroll-position-vertical-lr.html (and rl) fail in chromium (with "experimental web platform features" enabled). Can you report a bug if there isn't one already?

The 2dcontext tests I think are generated from tools/tests.yaml or some such. Can the new tests be there also?

nanto committed:
Revert "scrollPathIntoView method of CanvasRenderingContext2D"

This reverts commit e6efe05901189e1e2560a4b086d67398bab45a58.

nanto committed:
scrollPathIntoView method of CanvasRenderingContext2D

Almost same as commit e6efe05901189e1e2560a4b086d67398bab45a58 but genreted from tests2d.yaml.

nanto commented:
> html/browsers/browsing-the-web/scroll-to-fragid/scroll-position-vertical-lr.html (and rl) fail in chromium (with "experimental web platform features" enabled). Can you report a bug if there isn't one already?

I reported for Blink and Gecko.

* https://bugs.chromium.org/p/chromium/issues/detail?id=844868
* https://bugzilla.mozilla.org/show_bug.cgi?id=1462916

> The 2dcontext tests I think are generated from tools/tests.yaml or some such. Can the new tests be there also?

OK. I changed `tests2d.yaml`.

fserb reviewed (approved):
lgtm
